app/register.tsx

```diff
@@ -47,10 +53,70 @@ export default function Register() {
 
   const watchedUsername = watch("username");
 
+  // Check if user is already authenticated - if so, redirect to main app
+  useEffect(() => {
+    if (user) {
+      router.replace('/');
+      return;
+    }
+
+    // If user has existing accounts but none selected, show welcome screen
+    if (users.length > 0 && !user) {
+      router.replace('/welcome');
+      return;
+    }
+  }, [user, users, router]);
+
+  // Check invite code on component mount (only for unauthenticated users)
+  useEffect(() => {
+    // Skip invite validation if user is already authenticated
+    if (user || users.length > 0) return;
+
+    const checkInviteCode = async () => {
+      // First check if we have a previously validated invite code
+      const storedValidCode = getValidatedInvite();
+      if (storedValidCode) {
+        console.log('Found previously validated invite code');
+        setInviteCodeStatus('valid');
+        return;
+      }
+
+      // If there's a code in URL, validate it
+      if (code) {
+        setInviteCodeStatus('checking');
+        try {
+          const response = await validateInviteCode(code);
```

Comment:
@LiorAgnin invite code validation is not needed.

Invite-only access flow:
1. During signup, send the invite code with `generateRegistrationOptions`.
2. On the backend, verify that the invite code:
    - Matches the one sent in the user's email, and
    - Has not been used yet.
3. After successful signup, the user can log in using their registered passkey.

Since passkey registration already requires validation, a user can't obtain a passkey without a valid invite. Therefore, extra invite code validation during login is redundant.

Reply:
@MusabShakeel576 But users that didn't get the invite code won't see the signup at all.

Reply:
@LiorAgnin if url param doesn't have invite code, user will see an error message

Reply:
Yes, users will see this screen
![image](https://github.com/user-attachments/assets/cc282c37-340b-4c86-9009-097d1c27ffc8)